graphql/util.go

```diff
@@ -383,3 +385,84 @@ func ConvertDBTasksToGqlTasks(tasks []task.Task, baseTaskStatuses BaseTaskStatus
+		if err := model.RestartVersion(version.Id, modifications.TaskIds, modifications.Abort, user.Id); err != nil {
+			return errors.Errorf("error restarting patch: %s", err), http.StatusInternalServerError
+		}
+	case "set_active":
+		if err := model.SetVersionActivation(version.Id, modifications.Active, user.Id); err != nil {
+			return errors.Errorf("error activating patch: %s", err), http.StatusInternalServerError
+		}
+		// abort after deactivating the version so we aren't bombarded with failing tasks while
+		// the deactivation is in progress
+		if modifications.Abort {
+			if err := task.AbortVersion(version.Id, task.AbortInfo{User: user.DisplayName()}); err != nil {
+				return errors.Errorf("error aborting patch: %s", err), http.StatusInternalServerError
+			}
+		}
+		if !modifications.Active && version.Requester == evergreen.MergeTestRequester {
+			if proj == nil {
+				projRef, err := model.FindOneProjectRef(version.Branch)
+				if err != nil {
+					return errors.Errorf("error getting project ref: %s", err), http.StatusNotFound
+				}
+				proj = projRef
+			}
+			_, err := commitqueue.RemoveCommitQueueItem(proj.Identifier,
+				proj.CommitQueue.PatchType, version.Id, true)
+			if err != nil {
+				return errors.Errorf("error removing patch from commit queue: %s", err), http.StatusInternalServerError
+			}
+		}
+	case "set_priority":
+		if proj == nil {
+			projRef, err := model.FindOneProjectRef(version.Branch)
+			if err != nil {
+				return errors.Errorf("error getting project ref: %s", err), http.StatusNotFound
+			}
+			proj = projRef
+		}
+		if modifications.Priority > evergreen.MaxTaskPriority {
+			requiredPermission := gimlet.PermissionOpts{
+				Resource:      proj.Identifier,
+				ResourceType:  "project",
+				Permission:    evergreen.PermissionTasks,
+				RequiredLevel: evergreen.TasksAdmin.Value,
+			}
+			if !user.HasPermission(requiredPermission) {
+				return errors.Errorf("Insufficient access to set priority %v, can only set priority less than or equal to %v", modifications.Priority, evergreen.MaxTaskPriority), http.StatusUnauthorized
+			}
+		}
+		if err := model.SetVersionPriority(version.Id, modifications.Priority); err != nil {
+			return errors.Errorf("error setting version priority: %s", err), http.StatusInternalServerError
+		}
+	default:
+		return errors.Errorf("Unrecognized action: %v", modifications.Action), http.StatusBadRequest
+	}
+	return nil, 0
+}
+
+// ModifyVersionHandler handles the boilerplate code for performing a modify version action, i.e. schedule, unschedule, restart and set priority
+func ModifyVersionHandler(ctx context.Context, dataConnector data.Connector, patchID string, modifications Modifications) error {
+	version, err := dataConnector.FindVersionById(patchID)
+	if err != nil {
+		return ResourceNotFound.Send(ctx, fmt.Sprintf("error finding version %s: %s", patchID, err.Error()))
+	}
+	user := route.MustHaveUser(ctx)
+	err, _ = ModifyVersion(*version, *user, nil, modifications)
```

Comment:
You're discarding the type of error, but wouldn't you want to (for example) respond with NotFound if this is 404?

Reply:
The way i understand gql server errors is that the gql error code, e.g. `ResourceNotFound`, is utilized to identify the type of error instead of the HTTP status code, e.g. 404. This is why every gql error returned from resolvers is one of the error defined error constants. I got this idea from the apollo server paradigm for error handling: https://www.apollographql.com/docs/apollo-server/data/errors/. 

Right now I believe all errors returned from GQL server actually have HTTP status of 200 but apollo client knows to treat the response as an error is `errors` field is present in server response.

Reply:
Right, what I meant is that the function returns an HTTP status code giving more info on what went wrong, so don't you want to return the appropriate GQL error, rather than always returning an internal error?

Reply:
hmmmm, i can include it! it will also make it more consistent with the rest of the errors

Reply:
tldr: i do not think adding the http status code is necessary
after looking around the resolvers i see that many resolvers do not include an http status code, which is normal in a graphql server. if this were a standalone graphql server then there would be no http status codes at all.

Reply:
I'm not talking about returning an HTTP status code, I'm talking about using the code returned by that function to figure out which of these GQL errors to return: https://github.com/evergreen-ci/evergreen/blob/master/graphql/errors.go#L13

Reply:
i think you missed this part: https://github.com/evergreen-ci/evergreen/pull/3563/files#diff-5dbeeb4052af055f331b62e3cb9390efR465-R476

Reply:
the error returned from `ModifyVersion` is used as the error message in the returned error. only the http status code is left out.

Reply:
I didn't - that part is why I made the comment. If there's an error you'll always send an internal error even if the ModifyVersion function says that the project is not found

Reply:
gotcha. i wrote a utility function that maps http status to gql error